rest/route/github.go

```diff
@@ -462,6 +462,14 @@ func (gh *githubHookApi) handleMergeGroupChecksRequested(event *github.MergeGrou
 		"head_sha": event.GetMergeGroup().GetHeadSHA(),
 		"message":  "merge group received",
 	})
+	// Ensure that a project exists before creating an intent. Otherwise, intent creation will fail.
+	projectRefs, err := model.FindMergedEnabledProjectRefsByRepoAndBranch(org, repo, branch)
+	if err != nil {
+		return gimlet.NewJSONInternalErrorResponse(errors.Wrap(err, "finding project ref"))
+	}
+	if len(projectRefs) == 0 {
+		return gimlet.NewJSONInternalErrorResponse(errors.New("no matching project ref"))
```

Comment:
did we re-test this change? This still seems like the thing we don't want to do -- it's fine if the project isn't enabled and doesn't have the merge queue integrated, it's just if the project is enabled, we should send a comment. (honestly it's probably ideal to comment rather than send a status, since one could argue that someone should be allowed to use Evergreen for testing but not merging).

I'm actually also realizing I'm still confused about the leafygreen case, since we never used to get "Evergreen error" for them, and presumably were adding an intent, since we weren't running any checks to make sure we shouldn't do that. Should we just rollback revert the changes for this ticket first and see if that solves their problem, and then re-visit the solution? 

Sorry I know there's been a lot of back-and-forth about how we do this -- I think it is a valuable thing for users still, but is maybe becoming difficult so breaking it down could be good. I'm happy to have a pairing session or something if that'd be helpful post reverting, if we go that way.

Reply:
> it's fine if the project isn't enabled and doesn't have the merge queue integrated

This find above isn't checking for a project with the merge queue, it's checking for an enabled Evergreen project altogether (let me know if there is a preferable method for this). The intent itself ensures that the merge queue is enabled and sends a GitHub response if not; here, we want to totally abort if there is no applicable Evergreen project.

> I'm actually also realizing I'm still confused about the leafygreen case, since we never used to get "Evergreen error" for them, and presumably were adding an intent.

This is not the case. See in the [original PR](https://github.com/evergreen-ci/evergreen/pull/8290/files#diff-aba56b5b0ae44e07e6fbacc27ee9e5cb4bd1fb4d809263b6d13f2d55eee00ab7); we had checks that prevented the intent from being created.

Reply:
OHH i understand. Thank you for the explanation!